### PR TITLE
Phalcon\Security improvements

### PR DESCRIPTION
**Pull request by makerlabs**
- `Phalcon\Security` is using now `Phalcon\Security\Random`
- Enforced that `Phalcon\Security::getToken()` and `Phalcon\Security::getTokenKey()` return a random value per request not per call
- `Phalcon\Security::getToken()` and `Phalcon\Security::getTokenKey()` are using now `Phalcon\Security::_numberBytes` instead of passed as a argument or hardcoded value
- `Phalcon\Security::hash()` corrected not working `CRYPT_STD_DES`, `CRYPT_EXT_DES, MD5`, `CRYPT_SHA256`
- `Phalcon\Security::hash()` `CRYPT_SHA512` fixed wrong salt length
- Added missing unit-tests for `Phalcon\Security`

### CODE REVIEW

File: phalcon/security.zep [hunk 1]
```diff
@@ -79,6 +84,14 @@ class Security implements InjectionAwareInterface
 	const CRYPT_SHA512	   =	9;
 
 	/**
+	 * Phalcon\Security constructor
+	 */
+	public function __construct()
+	{
+		let this->_random = new Random();
```
Comment (sergeyklay) [inline phalcon/security.zep]: Could you please look at DI first?

**Comment by sergeyklay** [comment]
@makerlabs Could you please rebase?

**Comment by makerlabs** [comment]
@sergeyklay Ive done the rebase the only conflict was the changelog file but on PHP 5.4 travis is reporting a failed test in FileTest.php

**Comment by Jurigag** [comment]
Try to close and reopen it.

**Comment by makerlabs** [comment]
@Jurigag  thx it works now

**Comment by makerlabs** [comment]
@sergeyklay rebased

**Comment by sergeyklay** [comment]
Thanks

**Pull request merged**